### ISSUE TITLE
the `optimizer.zero_grad()` is called after loss backward

### ISSUE THREAD

**Issue by Dawn-LX** [cloneofsimo/lora#263]
I'm following your code to train LoRA for DreamBooth, but here is a bug in the training loop.

https://github.com/cloneofsimo/lora/blob/bdd51b04c49fa90a88919a19850ec3b4cf3c5ecd/training_scripts/train_lora_dreambooth.py#L888

the `optimizer.zero_grad()` should be called before `accelerator.backward(loss)`.
